Automattic/wp-calypso
# Jetpack product installer: Improve error logging

#### Changes proposed in this Pull Request

* Add status to error tracks event
* ~Add logstash for error~
* ~Rewrite `logToLogstash` action in TypeScript and provide interface for params according to PCYsg-5T4-p2~

Logstash requires backend changes. Moving those to another PR.

#### Testing instructions

I don't know how to hit this case organically. Here's a patch to force it when plugin install completes successfully:

```patch
diff --git a/client/my-sites/plans/current-plan/jetpack-product-install/index.tsx b/client/my-sites/plans/current-plan/jetpack-product-install/index.tsx
index eb21c18be7..4bc957a22f 100644
--- a/client/my-sites/plans/current-plan/jetpack-product-install/index.tsx
+++ b/client/my-sites/plans/current-plan/jetpack-product-install/index.tsx
@@ -277,7 +277,7 @@ export class JetpackProductInstall extends Component< Props, State > {
 		const hasErrorInstalling =
 			! this.shouldRefetchInstallationStatus() && this.installationHasRecoverableErrors();
 
-		if ( hasErrorInstalling && ! this.tracksEventSent ) {
+		if ( ( hasErrorInstalling || progressComplete === 100 ) && ! this.tracksEventSent ) {
 			this.tracksEventSent = true;
 			const { status } = this.props;
 
```

- Apply the above patch
- Purchase a plan for a Jetpack site without Akismet and VaultPress installed.
- Wait for successful completion
- The patch ensures we'll log on error.
- Watch the network for a logstash request. Check logstash logs.
